contracts/v2/PolygonZkEVMGlobalExitRootV2L2.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.8.20;
+import "../interfaces/IBasePolygonZkEVMGlobalExitRoot.sol";
+import {PolygonAccessControlUpgradeable} from "./lib/PolygonAccessControlUpgradeable.sol";
+
+/**
+ * Contract responsible for managing the exit roots for the L2 and global exit roots
+ * The special zkRom variables will be accessed and updated directly by the zkRom
+ */
+contract PolygonZkEVMGlobalExitRootV2L2 is
+    PolygonAccessControlUpgradeable,
+    IBasePolygonZkEVMGlobalExitRoot
+{
+    bytes32 internal constant _GLOBAL_EXIT_ROOT_SETTER_ROLE =
+        keccak256("GLOBAL_EXIT_ROOT_SETTER_ROLE");
+    bytes32 internal constant _GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN =
+        keccak256("GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN");
+
+    // Store every global exit root: Root --> timestamp
+    mapping(bytes32 => uint256) public globalExitRootMap;
+
+    // Rollup exit root will be updated for every PolygonZkEVMBridge call
+    bytes32 public lastRollupExitRoot;
+
+    // PolygonZkEVM Bridge address
+    address public immutable bridgeAddress;
+
+    /**
+     * @param _bridgeAddress PolygonZkEVMBridge contract address
+     */
+    constructor(address _bridgeAddress) {
+        bridgeAddress = _bridgeAddress;
+        _setRoleAdmin(
+            _GLOBAL_EXIT_ROOT_SETTER_ROLE,
+            _GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN
+        );
+        _setRoleAdmin(
+            _GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN,
+            _GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN
+        );
+        _grantRole(_GLOBAL_EXIT_ROOT_SETTER_ROLE_ADMIN, msg.sender);
```

Comment:
hmm tbh not sure I like having msg.sender be the admin - why not have it passed as a parameter in constructor?

but if @pgebheim is ok, then ok

Reply:
Oh yeah I think I agree..

@simondos is starting to look at this repo overall and should give the 👍

Reply:
Agree with @omnifient here, should be a parameter so we can set it to a multisig i.e.